storage/rocksdb/ha_rocksdb.cc

```diff
@@ -4149,6 +4149,41 @@ static int rocksdb_init_func(void *const p) {
 
   rocksdb_db_options->delayed_write_rate = rocksdb_delayed_write_rate;
 
+  // Check whether the filesystem backing rocksdb_datadir allows O_DIRECT
+  if (rocksdb_db_options->use_direct_reads &&
```

Comment:
Would it be better to move this after the check for use_direct_reads and allow_mmap_reads around line 4206? It would remove one condition from this if.